src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala

```diff
@@ -1,180 +1,166 @@
-          } else {
-            entity(as[AgoraEntity]) { newEntity =>
-              AgoraEntity.validate(newEntity) match {
-                case Success(_) => //noop
-                case Failure(errors) => throw new IllegalArgumentException(s"Method is invalid: Errors: $errors")
+    matchQuerySingleRoute(path)(ec) {
+      (namespace, name, snapshotId, username) =>
+        parameters( "onlyPayload".as[Boolean] ? false, "payloadAsObject".as[Boolean] ? false ) {
+          (onlyPayload, payloadAsObject) =>
+            val targetEntity = AgoraEntity(Option(namespace), Option(name), Option(snapshotId))
+            val entityTypes = AgoraEntityType.byPath(path)
+
+            get {
+              val queryAttempt = agoraBusiness.findSingle(targetEntity, entityTypes, username)
+
+              onComplete(queryAttempt) {
+                case Success(foundEntity) =>
+                  (onlyPayload, payloadAsObject) match {
+                    case (true, true) => complete(BadRequest, "onlyPayload, payloadAsObject cannot be used together")
+                    case (true, false) => complete(foundEntity.payload)
+                    case (false, true) => complete(foundEntity.withDeserializedPayload)
+                    case _ => complete(foundEntity)
+                  }
+                case Failure(error) => failWith(error)
               }
-              parameters("redact".as[Boolean] ? false) { redact =>
-                requestContext => completeEntityCopy(requestContext, targetEntity, newEntity, redact, username, path, queryHandlerProps)
+            } ~
+            delete {
+              val deletionAttempt = agoraBusiness.delete(targetEntity, entityTypes, username)
+
+              onComplete(deletionAttempt) {
+                case Success(rowsDeleted) => complete("FIX ME")
+                // TODO[GAWB-3052] The line above should be the line below; figure out what implicit may be missing
+                // case Success(rowsDeleted) => complete(rowsDeleted)
+                case Failure(error) => failWith(error)
+              }
+            } ~
+            post {
+              // only allow copying (post) for methods
+              if (path != AgoraConfig.methodsRoute) {
+                reject(MethodRejection(GET), MethodRejection(DELETE))
+              } else {
+                entity(as[AgoraEntity]) { newEntity =>
+                  AgoraEntity.validate(newEntity) match {
+                    case SuccessZ(_) => //noop
+                    case FailureZ(errors) => throw new IllegalArgumentException(s"Method is invalid: Errors: $errors")
+                  }
+                  parameters("redact".as[Boolean] ? false) { redact =>
+                    val copyingAttempt = agoraBusiness.copy(targetEntity, newEntity, redact, entityTypes, username)
+
+                    onComplete(copyingAttempt) {
+                      case Success(entities) => complete(entities)
+                      case Failure(error) => failWith(error)
```

Comment:
We may not need to handle the `Failure(error)` `case`s here since we have a top-level exception handler in `ApiService`.

Reply:
👍 A lot of the `onComplete ... Success ... Failure` conversion code was written when exception handling was not solidified and we needed it to make tests pass. Since exception handling was improved, we can simplify.